packages/gestalt/src/Button.js

```diff
@@ -28,54 +29,67 @@ const DEFAULT_TEXT_COLORS = {
   white: 'darkGray',
 };
 
-const SIZE_NAME_TO_PIXEL = {
-  sm: 10,
-  md: 12,
-  lg: 12,
-};
-
-type Props = {|
-  accessibilityControls?: string,
-  accessibilityExpanded?: boolean,
-  accessibilityHaspopup?: boolean,
-  accessibilityLabel?: string,
+type BaseButton = {|
   color?: 'gray' | 'red' | 'blue' | 'transparent' | 'white',
   disabled?: boolean,
   iconEnd?: $Keys<typeof icons>,
   inline?: boolean,
   name?: string,
-  onClick?: AbstractEventHandler<SyntheticMouseEvent<HTMLButtonElement>>,
-  selected?: boolean,
+  onClick?: AbstractEventHandler<
+    | SyntheticMouseEvent<HTMLButtonElement>
+    | SyntheticMouseEvent<HTMLAnchorElement>
+    | SyntheticKeyboardEvent<HTMLAnchorElement>
+    | SyntheticKeyboardEvent<HTMLButtonElement>
+  >,
   size?: 'sm' | 'md' | 'lg',
   text: string,
   textColor?: 'white' | 'darkGray' | 'blue' | 'red',
-  type?: 'submit' | 'button',
 |};
 
+type ButtonType = {|
+  ...BaseButton,
+  accessibilityControls?: string,
+  accessibilityExpanded?: boolean,
+  accessibilityHaspopup?: boolean,
+  accessibilityLabel?: string,
```

Comment:
@AlbertCarreras `accessibilityLabel?: string` should be moved to `BaseButton`. It's totally fine to set a label on a `Link` but not required.

We can also add it to `Link` if that prop isn't on that component yet.